packages/extension-base/src/page/PostMessageProvider.ts

```diff
@@ -165,10 +162,6 @@ export default class PostMessageProvider implements InjectedProvider {
   public async unsubscribe (type: string, method: string, id: number): Promise<boolean> {
     const subscription = `${type}::${id}`;
 
-    // FIXME This now could happen with re-subscriptions. The issue is that with a re-sub
-    // the assigned id now does not match what the API user originally received. It has
-    // a slight complication in solving - since we cannot rely on the send id, but rather
-    // need to find the actual subscription id to map it
```

Comment:
Couldn't figure out what is the issue here, I don't think we are interested in changing this, but if you could take a look @jalooc I'd be grateful

Reply:
It's possible that this problem does not exist any more because the subscriptiopn id is _saved_ in the clojure [here](https://github.com/Cardinal-Cryptography/aleph-zero-signer/blob/af62188f13ef3663c8fe38d6e7acc8ccf205e8dc/packages/extension-base/src/background/handlers/Tabs.ts#L159-L164) and therefore seems to be always relevant to the subscription being handled (opened -> closed). However it's impossible for me to say for sure: the flow is super complicated with classes and providers jumping between each other in spite of all having single, not replaced instances defined in the same codebase and on the same side of the communication channel; and the type casts don't help: [this](https://github.com/Cardinal-Cryptography/aleph-zero-signer/blob/c71492a5f41e78056d9e9658ef808fe2eb32d40e/packages/extension-base/src/page/PostMessageProvider.ts#L158) return type `number` is forced through [this](https://github.com/Cardinal-Cryptography/aleph-zero-signer/blob/c71492a5f41e78056d9e9658ef808fe2eb32d40e/packages/extension-base/src/page/PostMessageProvider.ts#L116) intermediary method returning nothing less than `any`, which I can only imagine is an already extracted value from the object being the actual return value from an rpc call. The lead stops here, but I'm assuming the _subscribe_ call listener does return a proper subscription id.

So my conclusion is that it's probably safe to deleted not only this comment, but also the entire `if` clause. But can't be sure without sacrificing a couple hours more to read through all those flow hoops.

Reply:
> So my conclusion is that it's probably safe to deleted (...) the entire if clause

Not that I'm recommending it either 😁